src/de/string_cache.rs

```diff
@@ -1,22 +1,37 @@
 use std::borrow::Cow;
 
-const LIMIT: usize = 1024;
+const LIMIT: usize = 1024; // Smile spec: maximum 1024 cached strings
 
 pub struct StringCache<'de> {
     vec: Vec<Cow<'de, str>>,
+    count: usize,  // Track insertion count for Jackson-compatible wrap-around
 }
 
 impl<'de> StringCache<'de> {
     pub fn new() -> Self {
-        StringCache { vec: vec![] }
+        StringCache { 
+            vec: Vec::with_capacity(LIMIT),
```

Comment:
We should leave this as is - it wastes a ton of memory in the common case where there are only a few cached values.